SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestNetReference/SpatialTestNetReference.cpp

```diff
@@ -171,9 +167,7 @@ void ASpatialTestNetReference::BeginPlay()
 				TArray<AActor*> CubesWithReferences;
 				UGameplayStatics::GetAllActorsOfClass(GetWorld(), ACubeWithReferences::StaticClass(), CubesWithReferences);
 
-				checkf(CubesWithReferences.Num() != 0, TEXT("There should never be 0 visible cubes"))
```

Comment:
Any reason for removing the checks?

Reply:
I thought that those might be unnecessary since the test is going to fail even in those cases checked. However, it can be argued that they bring some more 'safety' in terms of using robust UE mechanisms for checking the expected behaviour.   Do you think they bring value and should have been kept?

Reply:
I think I am fine with either. The way I think about it, I would soft-assert for things the test is focused on, like the number of references on the cubes, and check on things that the framework is relying on, e.g. the actor spawning, or an invariant, e.g. there should never be more than 4 cubes (kind of unnecessary in this case, just an example).

I guess from this standpoint, this particular check is not really needed, as yes, it is covered by the soft assertions below.
But in general, I think checks are cheap on a mental level when parsing the code, so usually in favor of adding them.